arch/xtensa/soc/esp32/esp32-mp.c

```diff
@@ -0,0 +1,190 @@
+#include <zephyr.h>
+#include <kernel_structs.h>
+#include <soc.h>
+
+#define _REG(base, off) (*(volatile u32_t *)((base) + (off)))
+
+#define RTC_CNTL_BASE             0x3ff48000
```

Comment:
`<soc/soc.h>` from esp-idf defines `DR_REG_RTCCNTL_BASE` for this register. I'd prefer to use it (and related macros) rather than defining our own, for consistency sake.  (Same for `DPORT_BASE`, etc.)